health_vara.py

```diff
@@ -257,6 +310,47 @@ def get_most_recent_imaging_request_datetime(self, _name):
             dt = max(i.date for i in self.imaging_test_requests)
             return dt
 
+    @classmethod
+    def search_most_recent_imaging_request_datetime(cls, _name, clause):
+        name, op, value = clause
+
+        patient = cls.__table__()
+
+        ImagingTestRequest = Pool().get('gnuhealth.imaging.test.request')
+        imaging_test_requests = ImagingTestRequest.__table__()
+
+        patient_with_requests = patient.join(
```

Comment:
I think combining multiple calls to SQL methods would make the query more readable here because then one can easily mentally convert it to the corresponding SQL code: 

```python
patient_with_latest_appointment = patient.join(
    imaging_test_requests,
    'LEFT',
    condition=patient.id == imaging_test_requests.patient
).select(patient.id,
         where=datetime_sql_where_clause(op, value, Max(
             imaging_test_requests.date).as_('max_date'))
         group_by=patient.id)

return [('id', 'in', patient_with_latest_appointment)]
```

Reply:
Hmm, I agree with the idea, I will see what I can do to simplify.

Unfortunately it's not going to be quite as nice as your suggestion. You can't use an aggregation (MAX) in a where clause, meaning I must use a subquery. And this library seems to forces me to have a reference to the table (in this case the subquery) in order to select columns.

Reply:
Maybe we can use a HAVING clause here in this case.

```python
patient_with_latest_appointment = patient.join(
    imaging_test_requests,
    'LEFT',
    condition=patient.id == imaging_test_requests.patient
).select(patient.id,
         having=datetime_sql_where_clause(op, value, Max(
             imaging_test_requests.date).as_('max_date'))
         group_by=patient.id)

return [('id', 'in', patient_with_latest_appointment)]
```

Reply:
Excellent idea, that has much improved the readability, thanks!